src/poll.rs

```diff
@@ -21,10 +21,24 @@ pub struct PollFd {
 impl PollFd {
     /// Creates a new `PollFd` specifying the events of interest
     /// for a given file descriptor.
-    pub const fn new(fd: RawFd, events: PollFlags) -> PollFd {
+    //
+    // Different from other I/O-safe interfaces, here, we have to take `AsFd`
+    // by reference to prevent the case where the `fd` is closed but it is
+    // still in use. For example:
+    //
+    // ```rust
+    // let (reader, _) = pipe().unwrap();
```

Comment:
Taking it by reference isn't sure proof against the file descriptor being closed prematurely.  Maybe we should add a `_fd: PhantomData<BorrowedFd<'a>>` to the struct?

Reply:
What about defining struct pollfd in Rust instead of using the libc one, just like [what rustix did](https://docs.rs/rustix/latest/rustix/io/struct.PollFd.html#).

With this, we can also resolve the next problem

Reply:
I'd rather not.  It's too easy to screw it up , especially if the structure is OS-dependent.  I assume you're suggesting using a BorrowedFd in the struct instead of a `c_int`?  Would that work any better than using PhantomData?

Reply:
> It's too easy to screw it up , especially if the structure is OS-dependent. 

This makes sense.

Adding a phantom field is a better approach.